proposals/1125_dynamic-supervisor-oidc-clients/README.md

```diff
@@ -0,0 +1,532 @@
+- `allowedGrantTypes`: May only contain the following valid options:
+    - `authorization_code` allows the client to perform the authorization code grant flow, i.e. allows the webapp to
+      authenticate users.  This grant must always be listed.
+    - `refresh_token` allows the client to perform refresh grants for the user to extend the user's session.  This grant
+      must be listed if `allowedScopes` lists `offline_access`.
+    - `urn:ietf:params:oauth:grant-type:token-exchange` allows the client to perform RFC8693 token exchange, which is a
+      step in the process to be able to get a cluster credential for the user.  This grant must be listed if
+      `allowedScopes` lists `pinniped:request-audience`.
+- `allowedScopes`: Decide what the client is allowed to request. Note that the client must also actually request
+  particular scopes during the authorization flow for the scopes to be granted. May only contain the following valid
+  options:
+    - `openid`: The client is allowed to request ID tokens.  ID tokens only include the
+      [required claims](https://openid.net/specs/openid-connect-core-1_0.html#IDToken) by default (`iss`, `sub`, `aud`,
+      `exp`, `iat`).  This scope must always be listed.
+    - `offline_access`: The client is allowed to request an initial refresh token during the authorization code grant
+      flow.  This scope must be listed if `allowedGrantTypes` lists `refresh_token`.
+    - `pinniped:request-audience`: The client is allowed to request a new audience value during a RFC8693 token
+      exchange, which is a step in the process to be able to get a cluster credential for the user.  `openid`,
+      `username` and `groups` scopes must be listed when this scope is present.  This scope must be listed if
+      `allowedGrantTypes` lists `urn:ietf:params:oauth:grant-type:token-exchange`.
+    - `username`: The client is allowed to request that ID tokens contain the user's username.  This is a newly
+      proposed scope which does not currently exist in the Supervisor. Without the `username` scope being requested and
+      allowed, the ID token would not contain the user's username.
+    - `groups`: The client is allowed to request that ID tokens contain the user's group membership, if their group
+      membership is discoverable by the Supervisor. This is a newly proposed scope which does not currently exist in the
+      Supervisor. Without the `groups` scope being requested and allowed, the ID token would not contain groups.
+- `phase`: This enum (`Pending`,`Ready`,`Error`) summarizes the overall status of the client (defaults to `Pending`).
+- `totalClientSecrets`: The number of client secrets that are currently associated with this client.
+- `conditions`: The result of validations performed by a controller on these CRs will be written by the controller on
+  the status. And example condition is showed above, and other conditions will also be added by the controller.
```

Comment:
```suggestion
  the status. An example condition is showed above, and other conditions will also be added by the controller.
```

Reply:
Updated in 7450fb6c8e9937e43fd5f0f95f2caa674413e3eb.